src/bin/rustfmt.rs

```diff
@@ -9,17 +9,37 @@
 // except according to those terms.
 
 #![cfg(not(test))]
+#![feature(path_ext)]
 
 extern crate rustfmt;
 
 use rustfmt::{WriteMode, run};
 
-use std::fs::File;
-use std::io::Read;
+use std::env;
+use std::fs::{File, PathExt};
+use std::io::{self, Read, Error, ErrorKind};
+use std::path::PathBuf;
+
+fn lookup_config_file() -> io::Result<PathBuf> {
+    let mut current = try!(env::current_dir());
+    loop {
+        let config_file = current.join("default.toml");
+        if config_file.exists() {
+            return Ok(config_file);
+        } else {
+            current = match current.parent() {
+                // if the current directory has no parent, we're done searching
+                None => return Err(Error::new(ErrorKind::NotFound, "config not found")),
+                Some(path) => path.to_path_buf(),
```

Comment:
I assume you meant better to assign `current` here, but I can't since `current.parent()` borrows `current` and you can't mutate something that's borrowed.